templates/terraform/modules/environment/main.tf

```diff
@@ -12,12 +12,22 @@ locals {
       ]
     }
   ]
+ 
+  user_access_roles = [
+    for r in var.roles : {
+      name         = r.name
+      aws_policy   = r.aws_policy
+      k8s_policies = r.k8s_policies
+    }
+  ]
 
-  kubernetes_iam_role_mapping = [
+  eks_kubernetes_iam_role_mapping = [
     for r in module.user_access.eks_iam_role_mapping : {
       iam_role_arn  = r.arn
       k8s_role_name = r.name
-      k8s_groups    = [ r.name ]
+      k8s_groups    = flatten(concat([r.name], [
+        for o in var.roles : o.k8s_groups if r.name == "${var.project}-kubernetes-${o.name}-${var.environment}"
```

Comment:
What's with this condition? Shouldn't we just be always merging `k8s_groups` with `r.name` ?

Reply:
combining 'module.user_access.eks_iam_role_mapping' and 'var.roles' to set 'k8s_groups'